src/Compiler/RefC/RefC.idr

```diff
@@ -635,7 +621,53 @@ mutual
         emit emptyFC "}"
         pure switchReturnVar
 
-    cStatementsFromANF (APrimVal fc c) _ = pure $ cConstant c
+    cStatementsFromANF (APrimVal fc (I x)) tailPosition = cStatementsFromANF (APrimVal fc (I64 $ cast x)) tailPosition
+    cStatementsFromANF (APrimVal fc c) _ = do
+      constdefs <- get ConstDef
+      case lookup c constdefs of
+           Just constid => constantName c constid -- the constant already booked.
+           Nothing => dyngen
+     where
+        constantName : Constant -> String -> Core String
+        constantName c n = case c of
+           I x   => pure "((Value*)&idris2_constant_Int64_\{cCleanString $ show x})"
+           I64 x => pure "((Value*)&idris2_constant_Int64_\{cCleanString $ show x})"
+           B64 x => pure "((Value*)&idris2_constant_Bits64_\{show x})"
+           Db x  => pure "((Value*)&idris2_constant_Double_\{cCleanString $ show x})"
+           Str x => pure "((Value*)&idris2_constant_String_\{n})"
+           _ => throw $ InternalError "[refc] Unsupported type of constant."
```

Comment:
this runtime error feels like it could be fairly easily remediated by a new smaller data-type than `Constant` so that we aren't relying on the invariant that no new type of constant is stored in `ConstDef` without a new case being added here.

Reply:
I pushed the changes.

I hope I understand the problem correctly. Anyway, this eliminated the runtime error